$OpenBSD: patch-src_nvim_getchar_c,v 1.1 2019/06/06 16:35:41 edd Exp $

Security patch: Source command doesn't check for the sandbox.
https://github.com/neovim/neovim/pull/10082

Detailed description:
https://github.com/numirias/security/blob/master/doc/2019-06-04_ace-vim-neovim.md

Index: src/nvim/getchar.c
--- src/nvim/getchar.c.orig
+++ src/nvim/getchar.c
@@ -1244,6 +1244,13 @@ openscript (
     EMSG(_(e_nesting));
     return;
   }
+
+  // Disallow sourcing a file in the sandbox, the commands would be executed
+  // later, possibly outside of the sandbox.
+  if (check_secure()) {
+    return;
+  }
+
   if (ignore_script)
     /* Not reading from script, also don't open one.  Warning message? */
     return;
